api/policies/AuthPolicy.js

```diff
@@ -44,18 +47,30 @@ async function isTOTPValid(user, token) {
   if (index === -1) {
     logger.warn(
       `[AuthPolicy->isTOTPValid] Invalid backup code ${token}`,
-      { security: true },
+      {
+        security: true,
+        fail: true,
+      },
     );
     throw Boom.unauthorized('Invalid backup code !', 'totp');
   }
 
-  // remove backup code so it can't be reused
-  user.totpConf.backupCodes.slice(index, 1);
-  user.markModified('totpConf');
+  // Remove backup code so it can't be reused.
+  user.totpConf.backupCodes.splice(index, 1);
+  user.markModified('totpConf.backupCodes');
```

Comment:
@orakili second block of code. most everything else is either for Lisa or logging tweaks